include/mbedtls/check_config.h

```diff
@@ -491,7 +491,7 @@
     defined(MBEDTLS_SHA256_C) || \
     defined(MBEDTLS_SHA384_C) || \
     defined(MBEDTLS_SHA512_C) || \
-    (defined(MBEDTLS_PSA_CRYPTO_C) && \
+    (defined(MBEDTLS_PSA_CRYPTO_CLIENT) && \
```

Comment:
I've simplified this block in https://github.com/Mbed-TLS/mbedtls/pull/8802 so you no longer need to make any change here.